buzz_el/graph_loader/knowledge_graph.py

```diff
@@ -0,0 +1,265 @@
+from os import PathLike
+from typing import Dict, Iterable, List, Optional, Set
+
+from rdflib import Graph
+from spacy.util import ensure_path
+
+from ..commons.utils import is_valid_url
+
+
+class KnowledgeGraph:
+    """
+    A class for loading a knowledge graph and interacting with it.
+
+    Parameters
+    ----------
+    kg_file_path : PathLike
+        The path to the knowledge graph file.
+    annotation_properties : Optional[Set[str]], optional
+        Set of RDF annotation properties used to extract entity labels, by default {"rdfs:label"}.
+    lang_filter : Optional[str], optional
+        Language filter tag to filter entity labels based on language, by default None.
+    interpretation_type: Enum["RDF", "RDFS", "OWL"], optional
+        Tag specifying how to interpret the KG file, by default "RDF".
+
+    Attributes
+    ----------
+    _kg_file_path : PathLike
+        The path to the knowledge graph file.
+    annotation_properties : Optional[Set[str]], optional
+        Set of RDF annotation properties used to extract entity labels, by default {"rdfs:label"}.
+    _annotation_properties: Set[str]
+        Annotation properties processed to be used directly in a SPARQL query.
+    _lang_filter : Optional[str], optional
+        Language filter tag to filter entity labels based on language, by default None.
+    _sparql_alternative_path_str: str
+        The portion of the SPARQL query constituting the alternative paths. 
+    _sparql_lang_filter_str: str
+        The portion of the SPARQL query constituting the language filter.
+    _interpretation_type: Enum["RDF", "RDFS", "OWL"], optional
+        Tag specifying how to interpret the KG file, by default "RDF".
+    kg: Graph
+        The KG object.
+    entity_patterns: List[Dict[str, str]]
+        The entity patterns.
+    """
+    def __init__(
+        self,
+        kg_file_path: PathLike,
+        annotation_properties: Optional[Set[str]] = None,
+        lang_filter_tag: Optional[str] = None,
+        interpretation_type: Optional[str] = None
+    ) -> None:
+        """Initialise the knowledge graph object.
+
+        Parameters
+        ----------
+        kg_file_path : PathLike
+            The path to the knowledge graph file.
+        annotation_properties : Optional[Set[str]], optional
+            Set of RDF annotation properties used to extract entity labels, by default {"rdfs:label"}.
+        lang_filter_tag : Optional[str], optional
+            Language filter tag to filter entity labels based on language, by default None.
+        interpretation_type : Optional[str], optional
+           Tag specifying how to interpret the KG file, by default "RDF".
+        """
+        self._kg_file_path = ensure_path(kg_file_path)
+
+        self._annotation_properties = {"rdfs:label"}
+        self.annotation_properties = annotation_properties
```

Comment:
I suppose that you are doing this to use the setter but I don't understand when the self._annotation_properties is set with self.annotation_properties

Reply:
When you do `self.annotation_properties = annotation_properties`, you are actually calling the setter, which sets `self._annotation_properties`.

Remember, a while back I played around with this (maybe too much, but I believe it is a good occasion)